app/scripts/modules/amazon/src/pipeline/stages/bake/awsBakeStage.js

```diff
@@ -85,6 +87,15 @@ module.exports = angular
 
           if (!$scope.stage.baseOs && $scope.baseOsOptions && $scope.baseOsOptions.length) {
             $scope.stage.baseOs = $scope.baseOsOptions[0].id;
+          } else if (
+            $scope.stage.baseOs &&
+            !($scope.baseOsOptions || []).find(baseOs => baseOs.id === $scope.stage.baseOs)
+          ) {
+            $scope.baseOsOptions.push({
+              id: $scope.stage.baseOs,
+              detailedDescription: 'Custom',
+              vmTypes: ['hvm', 'pv'],
```

Comment:
oh hey this is kind of a cool idea. only downside is it means we lose the ability to validate OS options that we remove because they're deprecated. but maybe that's ok.

Reply:
Well the current experience is terrible because it just JSEs.

If we wanted to aggressively validate against deprecated versions, I’d imagine we can return a “deprecated” blacklist or flag from the gate endpoint.

But not sure it’s worth that effort.

Reply:
yeah, i went through basically the same thought process — i think this is good enough until we decide we need some kind of deprecated indicator